spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/InstanceRegistry.java

```diff
@@ -59,20 +62,20 @@ public Mono<InstanceId> register(Registration registration) {
 	}
 
 	/**
-	 * Get a list of all registered instances.
-	 * @return list of all instances.
+	 * Get a list of all registered instances that satisfy the filter.
+	 * @return list of all instances satisfying the filter.
 	 */
 	public Flux<Instance> getInstances() {
-		return repository.findAll();
+		return repository.findAll().filter(filter::filter);
```

Comment:
Wouldn't it make more sense to filter during registration?

Reply:
No, as the issue was a feature request to allow filtering for parts of the application only (i.e. wallboard). Hence, we should not prevent registration of services, but allow to filter the result.